cabal-install/src/Distribution/Client/ProjectBuilding.hs

```diff
@@ -652,57 +656,64 @@ rebuildTargets
     | otherwise = do
         -- Concurrency control: create the job controller and concurrency limits
         -- for downloading, building and installing.
-        jobControl <-
-          if isParallelBuild
-            then newParallelJobControl buildSettingNumJobs
-            else newSerialJobControl
+        mkJobControl <- case buildSettingNumJobs of
+          Serial -> newSerialJobControl
+          NumJobs n -> newParallelJobControl (fromMaybe numberOfProcessors n)
```

Comment:
Same problem, is n=1 case handled correctly?

Reply:
Here yes. We could optimise this later when the system is better understood, but for now let's leave it as that.